### PR TITLE
FF96 Canvas supports exporting images as webp

### PR DESCRIPTION
**Pull request by hamishwillee**
FF96 adds a webp image encoder used by a number of canvas APIs - https://bugzilla.mozilla.org/show_bug.cgi?id=1511670

This updates the affected APIs where they differ from what the spec says and mentions the possibility of webp being supported. Note that the actual support for webp is going to indicated in BCD.

It also updates the image guide to note that webp is supported for image export. Wasn't so sure on that one. 

Other docs work for this can be tracked in #10853

### CODE REVIEW

File: files/en-us/web/api/htmlcanvaselement/toblob/index.md [hunk 1]
```diff
@@ -11,30 +11,32 @@ browser-compat: api.HTMLCanvasElement.toBlob
...
 - `callback`
   - : A callback function with the resulting {{domxref("Blob")}} object as a single argument.
+    `null` may be passed if the image cannot be created for any reason.
 - `type` {{optional_inline}}
   - : A {{domxref("DOMString")}} indicating the image format.
     The default type is `image/png`; that type is also used if the given type isn't supported.
 - `quality` {{optional_inline}}
-
-  - : A {{jsxref("Number")}} between `0` and `1`, indicating image quality if the requested type is `image/jpeg` or `image/webp`.
-    If this argument is anything else, the default values 0.92 and 0.80 are used for image/jpeg and image/webp respectively.
```
Comment (hamishwillee) [inline files/en-us/web/api/htmlcanvaselement/toblob/index.md:34]: Note, the spec just says "user agent defaults" are used if the value is outside the range. It also doesn't specifically  mention webp - quality is relevant generally to formats that are support lossy compression. I have changed this in all cases.

File: files/en-us/web/api/offscreencanvas/converttoblob/index.md [hunk 2]
```diff
@@ -27,15 +33,33 @@ OffscreenCanvas.convertToBlob(options)
...
 ### Return value
 
 A {{jsxref("Promise")}} returning a {{domxref("Blob")}} object representing the image contained in the canvas.
 
+### Exceptions
+
+The promise may be rejected with the following exceptions:
+
+- `InvalidStateError` {{domxref("DOMException")}}
+  - : The `OffscreenCanvas` is not detached; in other words it still associated with the DOM and not the current worker.
```
Comment (hamishwillee) [inline files/en-us/web/api/offscreencanvas/converttoblob/index.md:51]: Note, this was my understanding of the exception and it makes sense. But someone with better knowledge of offscreen canvas might want to confirm.

File: files/en-us/web/api/offscreencanvas/converttoblob/index.md [hunk 3]
```diff
@@ -27,15 +33,33 @@ OffscreenCanvas.convertToBlob(options)
...
+
+- `SecurityError` {{domxref("DOMException")}}
+  - : The canvas context mode is 2d and the bitmap is not origin clean; at least some of its contents have or may have been loaded from a site other than the one from which the document itself was loaded.
```
Comment (sideshowbarker) [inline files/en-us/web/api/offscreencanvas/converttoblob/index.md:54]:
```suggestion
  - : The canvas context mode is 2d and the bitmap is not origin-clean; at least some of its contents have or may have been loaded from a site other than the one from which the document itself was loaded.
```

File: files/en-us/web/api/htmlcanvaselement/toblob/index.md [hunk 4]
```diff
@@ -43,7 +45,7 @@ None.
 ### Exceptions
 
 - `SecurityError`
-  - : The canvas's bitmap is not origin clean; at least some of its contents come from secure
+  - : The canvas's bitmap is not origin clean; at least some of its contents have or may have been loaded from a site other than the one from which the document itself was loaded.
```
Comment (sideshowbarker) [inline files/en-us/web/api/htmlcanvaselement/toblob/index.md:48]:
```suggestion
  - : The canvas's bitmap is not origin-clean; at least some of its contents have or may have been loaded from a site other than the one from which the document itself was loaded.
```

File: files/en-us/web/api/offscreencanvas/converttoblob/index.md [hunk 5]
```diff
@@ -27,15 +33,33 @@ OffscreenCanvas.convertToBlob(options)
...
+
+- `IndexSizeError` {{domxref("DOMException")}}
+  - : The canvas bitmap has no pixels (either the horizontal or veritical dimension is zero).
```
Comment (sideshowbarker) [inline files/en-us/web/api/offscreencanvas/converttoblob/index.md:57]:
```suggestion
  - : The canvas bitmap has no pixels (either the horizontal or vertical dimension is zero).
```

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**